Library/Formula/p7zip.rb

```diff
@@ -18,16 +18,19 @@ def install
       mv 'makefile.macosx_32bits', 'makefile.machine'
     else
       mv 'makefile.macosx_64bits', 'makefile.machine'
-    end
+    end if OS.mac?
+
+    mv 'makefile.linux_any_cpu', 'makefile.machine' if OS.linux?
 
+    # install.sh chmods to 444, which is bad and breaks uninstalling
+    inreplace "install.sh" do |s|
+      s.gsub! /chmod 444.*|chmod 555.*/, ""
+    end
```

Comment:
`inreplace "install.sh", /^chmod .*/, ""` would be too general, because there are lines like:

`chmod 555 "${DEST_DIR}${DEST_SHARE}/${prg}"`